### ISSUE TITLE
Readonly columns can be bypassed by using string values as table/column references

### ISSUE THREAD

**Issue by henry99923123** [jOOQ/jOOQ#15710]
### Expected behavior

I have a book table with a column called 'title' which is configured as a read only column via code generator. The settings are set to throw if someone tries to insert a value for the title. I'm testing this and it works if I use proper references with Book.BOOK and Book.BOOK.TITLE. But if I pass in strings as table names and column names it should throw too.

The same should work for updates too, not just for insertions.

### Actual behavior

If I pass in strings as table / column names it doesn't throw and the user is allowed to insert values. The same behavior happens for updates too.

### Steps to reproduce the problem

Create a table called book in postgresql with fields: id, author_id and title. The code generator should set the title-column to readonly. The settings object for the jOOQ-Context should set everything to throw, like this:

```
this.settings.setReadonlyInsert(WriteIfReadonly.THROW);
this.settings.setReadonlyUpdate(WriteIfReadonly.THROW);
this.settings.setReadonlyTableRecordInsert(WriteIfReadonly.THROW);
this.settings.setReadonlyUpdatableRecordUpdate(WriteIfReadonly.THROW);
```

Then execute the following code:

```
context.insertInto(table(name("book")))
                .set(field(name("author_id")), 1)
                .set(field(name("title")), "any value")
                .execute();
```

And for an update:

```
context.update(Book.BOOK)
                .set(field(name("title")), UUID.randomUUID().toString())
                .where(Book.BOOK.ID.eq(1))
                .execute();
```

### jOOQ Version

jOOQ Professional Edition 3.18.7

### Database product and version

PostgreSQL 15

### Java Version

openjdk 17.0.8 2023-07-18

### OS Version

Ubuntu

### JDBC driver name and version (include name if unofficial driver)

org.postgresql:postgresql:42.6.0

**Comment by lukaseder** [comment]
Thanks for your message.

The recently implemented jOOQ 3.19 feature "policies" (https://github.com/jOOQ/jOOQ/issues/2682) has a dedicated section on this topic, given that it may be perceived as a security feature (which it isn't, really):

- https://www.jooq.org/doc/dev/manual/sql-building/queryparts/policies/policies-security/

It's definitely worth adding an additional section somewhere that discusses similar limitations of client side implementations of server side SQL features:

- https://github.com/jOOQ/jOOQ/issues/15712

In short, what you're expecting can't be done. The readonly property is attached to a column expression (its `DataType`, more specifically). If you just construct arbitrary expressions without such readonly information, then, there's simply no readonly information that applies to your query.

The question is, if you're using the code generator, then why not use the generated code?
